src/firstboot/firstboot.c

```diff
@@ -672,24 +672,22 @@ static int write_root_passwd(const char *passwd_path, const char *password, cons
                         return r;
 
         } else {
-                struct passwd root = {
-                        .pw_name = (char *) "root",
-                        .pw_passwd = (char *) password,
-                        .pw_uid = 0,
-                        .pw_gid = 0,
-                        .pw_gecos = (char *) "Super User",
-                        .pw_dir = (char *) "/root",
-                        .pw_shell = (char *) (shell ?: "/bin/sh"),
-                };
-
                 if (errno != ENOENT)
                         return -errno;
 
                 r = fchmod(fileno(passwd), 0644);
                 if (r < 0)
                         return -errno;
 
-                r = putpwent_sane(&root, passwd);
+                r = putpwent_sane(&(struct passwd) {
+                                .pw_name = (char *) "root",
+                                .pw_passwd = (char *) password,
+                                .pw_uid = 0,
+                                .pw_gid = 0,
+                                .pw_gecos = (char *) "Super User",
+                                .pw_dir = (char *) "/root",
+                                .pw_shell = (char *) (shell ?: "/bin/sh"),
+                        }, passwd);
```

Comment:
I'm a fan of structured initialization... but defining long structs like this inline, not so much. I'm not convinced that this is better.